entities/src/test/java/org/odk/collect/entities/javarosa/filter/LocalEntitiesFilterStrategyTest.kt

```diff
@@ -67,7 +67,7 @@ class LocalEntitiesFilterStrategyTest {
     }
 
     @Test
-    fun `returns matching nodes when entity matches name`() {
+    fun `returns matching nodes when id = value`() {
```

Comment:
Definitely my fault here, but now most expressions are optimized, it's hard to get a quick sense of tests that check for optimized expressions and ones that don't. Maybe a quick set of renames around the theme of expressions that do or don't fall through would help?

I think once we get to optimizing all secondary instances, we'll most likely want to rework the tests or pull out something that just handles the XPath to `Query` translation, but I think doing that now feels over the top (even though it'd probably be more focused).

Reply:
Something like this https://github.com/getodk/collect/pull/6539/commits/870c23dfc793a1f3ed7e7f0c56b37573988fa70a?